lib/net/ldap/connection.rb

```diff
@@ -88,6 +88,13 @@ def self.wrap_with_ssl(io, tls_options = {})
     conn = OpenSSL::SSL::SSLSocket.new(io, ctx)
     conn.connect
 
+    if tls_options[:verify_host]
+      # This raises OpenSSL::SSL::SSLError if hostname verification fails
+      conn.post_connection_check(tls_options[:verify_host])
```

Comment:
A previously undocumented (until I documented it) method it is very important to call if you want your SSL connections to be secure!